src/main/java/org/opengis/cite/iso19142/util/ServiceMetadataUtils.java

```diff
@@ -87,8 +88,25 @@ public static URI getOperationEndpoint(final Document wfsMetadata, String opName
         xpath.setNamespaceContext(nsBindings);
         URI endpoint = null;
         try {
-            String href = xpath.evaluate(expr, wfsMetadata);
-            endpoint = URI.create(href);
+            String pathToEvaluate = String.format("//ows:Operation[@name='%s']//ows:%s/ows:Constraint/ows:AllowedValues/ows:Value", opName, method.toString());
+            Object evalResult = xpath.evaluate(pathToEvaluate, wfsMetadata, XPathConstants.NODESET);
+            NodeList allowedValues = (NodeList) evalResult;
+
+            // To get SOAP end point on basis of allowed values
+            if (originalBinding.equals(ProtocolBinding.SOAP) && allowedValues.getLength() > 0) {
+                evalResult = xpath.evaluate(expr, wfsMetadata, XPathConstants.NODESET);
+                NodeList hrefList = (NodeList) evalResult;
+
+                for (int i = 0; i < hrefList.getLength(); i++) {
+                    if (allowedValues.item(i).getTextContent().equalsIgnoreCase("soap")) {
```

Comment:
What happens if "soap" is not part of allowedValues? Shouldn't the default behavior be executed then (ows:Post is used)?

Reply:
Yes, I will add a condition for that.